<?xml version="1.0"?>
<launch>
    <arg name="base_link_frame" default="base_link"/>
    <arg name="map_frame" default="map"/>
    <arg name="initial_pose_topic" default="initial_pose_topic"/>
    <arg name="twist_topic" default="/twist"/>
    <arg name="pose_topic" default="/pose"/>
    <arg name="use_2d_pose_estimate" default="false"/>
    <arg name="estimate_3d_pose" default="false"/>
    <arg name="update_rate" default="30"/>
    <arg name="num_particles" default="100"/>

    <node pkg="pf_localization" type="pf_localization_node" name="pf_localization_node" output="screen" respawn="false" respawn_delay="0">
        <param name="base_link_frame" value="$(arg base_link_frame)"/>
        <param name="map_frame" value="$(arg map_frame)"/>
        <param name="initial_pose_topic" value="$(arg initial_pose_topic)"/>
        <param name="pose_topic" value="$(arg pose_topic)"/>
        <param name="twist_topic" value="$(arg twist_topic)"/>
        <param name="estimate_3d_pose" value="$(arg estimate_3d_pose)"/>
        <param name="update_rate" value="$(arg update_rate)"/>
        <param name="num_particles" value="$(arg num_particles)"/>
    </node>
</launch>